src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -284,7 +299,19 @@ private String getInputArguments() {
             artifact.addFilePathArgTo(args);
         }
 
-        args.forEach((key, val) -> inputArgsList.add("'" + key + "'" + "," + "'" + val + "'"));
+        // Add source folder options to argument
+        SourceFolder sf = getSourceFolder();
+        if(sf != null && !sf.getSourceFolderPaths().isEmpty()){
+            sf.addFilePathArgTo(SOURCE_FOLDER, args);
+        }
+
+        args.forEach((key, val) -> {
+            if(key.equals(SOURCE_FOLDER)){

```

Comment:
Would it be good idea if Source folder knew what it adds as parameters to inputArgsList. for example 
 sf.addFilePathArgTo(inputArgsList) . the rationale is the source folder knows what it expects and it provides args in that way.

Reply:
Refactored the code.

Reply:
What is it doing differently than earlier ? am i missing something . As i mentioned sf.addFilepathArgTo() adds required arguments in List<> then you may not require the check for key in the forEach

Reply:
Hi Nikhil,

Could you check my last commit? I have made that change https://github.com/mathworks/jenkins-matlab-plugin/pull/148/commits/86a9bd281249c5aee8ba564aac8090df07ffed6d 

Best,
Anmol